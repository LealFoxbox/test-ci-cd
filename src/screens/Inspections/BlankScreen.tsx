import React from 'react';
import { Paragraph, Title } from 'react-native-paper';

import { PersistentUserStore } from 'src/pullstate/persistentStore';

import { BlankContentContainer } from './styles';

const DownloadingScreen: React.FC = () => {
  const userData = PersistentUserStore.useState((s) => s.userData);

  return (
    <BlankContentContainer>
      <Title>No Areas Found</Title>
      <Paragraph style={{ marginTop: 0 }}>You don't have any areas of access.</Paragraph>
      <Paragraph style={{ marginTop: 0 }}>Contact {userData?.account.name} for help.</Paragraph>
    </BlankContentContainer>
  );
};

export default DownloadingScreen;
