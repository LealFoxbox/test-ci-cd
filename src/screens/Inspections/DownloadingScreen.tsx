import React from 'react';
import { Paragraph, ProgressBar, useTheme } from 'react-native-paper';

import { DowloadingContainer } from './styles';

const DownloadingScreen: React.FC<{ progress: number }> = ({ progress }) => {
  const theme = useTheme();

  return (
    <DowloadingContainer>
      <ProgressBar progress={progress / 100} color={theme.colors.primary} />
      <Paragraph style={{ textAlign: 'center', marginTop: 20 }}>Downloading account data...</Paragraph>
    </DowloadingContainer>
  );
};

export default DownloadingScreen;
