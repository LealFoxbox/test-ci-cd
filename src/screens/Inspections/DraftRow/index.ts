export * from './DraftRow';
export { default } from './DraftRow';
