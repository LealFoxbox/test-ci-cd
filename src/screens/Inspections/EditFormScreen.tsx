import React from 'react';
import { View } from 'react-native';
import { RouteProp, useRoute } from '@react-navigation/native';

import { PersistentUserStore } from 'src/pullstate/persistentStore';
import { INSPECTIONS_FORM } from 'src/navigation/screenNames';
import { InspectionsNavigatorParamList } from 'src/navigation/InspectionsNavigator';
import FormScreen from 'src/components/FormScreen';
import { User } from 'src/types';
import { getBaseUrl } from 'src/services/api/utils';

function getFormUri(user: User, structureId: number, formId: number) {
  return `${getBaseUrl(
    user.account.subdomain,
  )}/inspect/areas/${structureId}/inspection_forms/${formId}?mobile_app=1&user_credentials=${user.single_access_token}`;
}

const EditFormScreen: React.FC<{}> = () => {
  const userData = PersistentUserStore.useState((s) => s.userData);
  const {
    params: { formId, structureId },
  } = useRoute<RouteProp<InspectionsNavigatorParamList, typeof INSPECTIONS_FORM>>();

  if (!userData || !formId || !structureId) {
    return <View />;
  }

  return <FormScreen source={{ uri: getFormUri(userData, structureId, formId) }} />;
};

export default EditFormScreen;
