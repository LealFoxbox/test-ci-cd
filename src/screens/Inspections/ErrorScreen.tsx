import React from 'react';
import { Button, Title } from 'react-native-paper';

import ConnectionBanner from 'src/components/ConnectionBanner';
import { styled } from 'src/paperTheme';
import { DownloadStore } from 'src/pullstate/downloadStore';
import { useNetworkStatus } from 'src/utils/useNetworkStatus';

const MessageContainer = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
  padding: 30px;
`;

const ErrorScreen: React.FC<{}> = () => {
  const connected = useNetworkStatus();

  return (
    <>
      <ConnectionBanner connected={connected} />
      <MessageContainer>
        <Title style={{ textAlign: 'center', marginBottom: 15 }}>
          An error has ocurred while downloading, please try again
        </Title>
        <Button
          mode="contained"
          disabled={!connected}
          dark
          accessibilityLabel="download"
          icon="cloud-download"
          onPress={() => {
            // this triggers a retry
            DownloadStore.update((s) => {
              s.progress = 0;
              s.error = null;
            });
          }}
        >
          Download
        </Button>
      </MessageContainer>
    </>
  );
};

export default ErrorScreen;
