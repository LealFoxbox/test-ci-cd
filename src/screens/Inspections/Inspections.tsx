import React from 'react';
import { Title } from 'react-native-paper';

import { PersistentUserStore } from 'src/pullstate/persistentStore';
import { DownloadStore } from 'src/pullstate/downloadStore';

import { Container, MessageContainer } from './styles';

const InspectionsScreen: React.FC<{}> = () => {
  const userData = PersistentUserStore.useState((s) => s.userData);
  const { progress, error } = DownloadStore.useState((s) => s);

  if (!userData) {
    return <Container />;
  }

  if (error) {
    return (
      <MessageContainer>
        <Title style={{ textAlign: 'center' }}>An error ocurred, please reload</Title>
      </MessageContainer>
    );
  }

  if (progress !== 100) {
    return (
      <MessageContainer>
        <Title style={{ textAlign: 'center' }}>Downloading, progress is {progress}</Title>
      </MessageContainer>
    );
  }

  return (
    <MessageContainer>
      <Title style={{ textAlign: 'center' }}>Job done, showing inspections from db, I swear.</Title>
    </MessageContainer>
  );
};

export default InspectionsScreen;
