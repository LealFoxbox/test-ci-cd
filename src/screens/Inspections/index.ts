export * from './Inspections';
export { default } from './Inspections';
