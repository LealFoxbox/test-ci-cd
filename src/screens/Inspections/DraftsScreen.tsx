import React from 'react';
import { Divider, useTheme } from 'react-native-paper';
import { useNavigation } from '@react-navigation/native';
import { FlatList } from 'react-native';
import { sortBy } from 'lodash/fp';

import { INSPECTIONS_FORM } from 'src/navigation/screenNames';
import { PersistentUserStore } from 'src/pullstate/persistentStore';
import { styled } from 'src/paperTheme';
import SwipableRow from 'src/components/SwipableRow';
import { deleteDraftAction } from 'src/pullstate/formActions';

import DraftRow from './DraftRow';

const Container = styled.View`
  flex: 1;
  background-color: ${({ theme }) => theme.colors.background};
  justify-content: center;
`;

const DraftsScreen: React.FC<{}> = () => {
  const theme = useTheme();
  const navigation = useNavigation();
  const drafts = PersistentUserStore.useState((s) => sortBy('started_at', Object.values(s.drafts)));

  return (
    <Container theme={theme}>
      <FlatList
        contentContainerStyle={{
          justifyContent: 'flex-start',
        }}
        data={drafts}
        ItemSeparatorComponent={Divider}
        renderItem={({ item }) => (
          <SwipableRow leftLabel="Delete draft" onPressLeft={() => deleteDraftAction(item.assignmentId)}>
            <DraftRow
              label={item.name}
              content={item.locationPath}
              onPress={() => {
                navigation.navigate(INSPECTIONS_FORM, {
                  formId: item.formId,
                  structureId: item.structureId,
                  assignmentId: item.assignmentId,
                  title: item.name,
                });
              }}
              hasPhotos={Object.values(item.fields).some((f) => f.photos.length > 0)}
              lastModified={item.lastModified || item.started_at || Date.now()}
            />
          </SwipableRow>
        )}
        keyExtractor={(item) => `${item.guid}`}
      />
    </Container>
  );
};

export default DraftsScreen;
