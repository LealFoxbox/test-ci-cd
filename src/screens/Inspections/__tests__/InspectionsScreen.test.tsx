import React from 'react';

import { act, render } from 'helpers/testUtils';

import Inspections from '../InspectionsScreen';

describe('InspectionsScreen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.useFakeTimers();
  });

  it('should render', async () => {
    const result = render(<Inspections />);

    // eslint-disable-next-line @typescript-eslint/no-misused-promises
    await act(() => Promise.resolve());

    expect(result).toMatchSnapshot();
  });
});
