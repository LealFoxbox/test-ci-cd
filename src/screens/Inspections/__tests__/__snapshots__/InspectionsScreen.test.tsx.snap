// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InspectionsScreen should render 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          collapsable={false}
          enabled={true}
          pointerEvents="auto"
          removeClippedSubviews={false}
          style={
            Array [
              Object {
                "flex": 1,
                "overflow": "hidden",
              },
              null,
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              accessibilityElementsHidden={false}
              importantForAccessibility="auto"
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#e9e9e9",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "justifyContent": "center",
                    },
                  ]
                }
                theme={
                  Object {
                    "animation": Object {
                      "scale": 1,
                    },
                    "colors": Object {
                      "accent": "#F2680E",
                      "backdrop": "rgba(0, 0, 0, 0.5)",
                      "background": "#e9e9e9",
                      "deficient": "#FF3333",
                      "disabled": "rgba(0, 0, 0, 0.26)",
                      "error": "#FF2D55",
                      "gps": "#007AFE",
                      "notification": "#FF9400",
                      "onBackground": "#000000",
                      "onSurface": "#FF9400",
                      "placeholder": "rgba(0, 0, 0, 0.54)",
                      "primary": "#FF9400",
                      "secondaryText": "#888",
                      "success": "#34C759",
                      "surface": "#FFFFFF",
                      "text": "#000000",
                    },
                    "dark": false,
                    "fonts": Object {
                      "light": Object {
                        "fontFamily": "System",
                        "fontWeight": "300",
                      },
                      "medium": Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      "regular": Object {
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      "thin": Object {
                        "fontFamily": "System",
                        "fontWeight": "100",
                      },
                    },
                    "roundness": 4,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "rgb(255, 255, 255)",
            "borderTopColor": "rgb(216, 216, 216)",
            "borderTopWidth": 0.5,
            "bottom": 0,
            "elevation": 8,
            "height": 49,
            "left": 0,
            "paddingBottom": 0,
            "paddingHorizontal": 0,
            "position": null,
            "right": 0,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityLabel="ComponentToTest, tab, 1 of 1"
            accessibilityRole="button"
            accessibilityState={
              Object {
                "selected": true,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "flexDirection": "column",
                  "justifyContent": "flex-end",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgb(0, 122, 255)",
                  },
                  Object {
                    "fontSize": 10,
                  },
                  undefined,
                ]
              }
            >
              ComponentToTest
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
